src/main/java/org/apache/aurora/scheduler/configuration/ConfigurationManager.java

```diff
@@ -402,6 +402,10 @@ public ITaskConfig validateAndPopulate(ITaskConfig config) throws TaskDescriptio
     if (config.isSetKillPolicy() && config.getKillPolicy().getGracePeriodSecs() < 0) {
       throw new TaskDescriptionException("Grace Period should be greater than 0");
     }
+    
+    if (config.isSetHealthCheck() && !config.getHealthCheck().isSetHttp() && !config.getHealthCheck().isSetShell()) {
```

Comment:
What if both health checks are set?

Reply:
You need both of them to set up a Healthcheck, right? Should the error message read http _and_ shell?